pynumaflow/batchmapper/servicer/async_servicer.py

```diff
@@ -0,0 +1,235 @@
+            async for msg in self.present_data(datum_iterator):
+                yield msg
+        except Exception as err:
+            _LOGGER.critical("UDFError, re-raising the error", exc_info=True)
+            raise err
+
+    async def present_data(
+        self, datum_iterator: AsyncIterable[Datum]
+    ) -> AsyncIterable[batchmap_pb2.BatchMapResponse]:
+        try:
+            async for response_msgs in self._handler(datum_iterator):
+                # Implicitly ensure we have a list
+                yield batchmap_pb2.BatchMapResponse(
+                    results=[
+                        batchmap_pb2.BatchMapResponse.Result(
+                            keys=msg.keys, value=msg.value, tags=msg.tags
+                        )
+                        for msg in response_msgs.messages
+                    ],
+                    id=response_msgs.id,
+                )
+        except Exception as err:
+            err_msg = "UDSinkError: %r" % err
+            _LOGGER.critical(err_msg, exc_info=True)
+
+            async for _datum in datum_iterator:
+                yield batchmap_pb2.BatchMapResponse(
+                    batchmap_pb2.BatchMapResponse.Result.as_failure(_datum.id, err_msg)
+                )
+
+
+# ----
+class BatchMapUnaryServicer(BatchMapServicer):
+    """Operates a async-mapper unary style prentation. Does not provide batching,
+    but a streamlined interface for individual message handling that can be used in place of `mapper.MapAsyncServer`
+    """
+
+    async def present_data(
+        self,
+        datum_iterator: AsyncIterable[Datum],
+    ) -> AsyncIterable[batchmap_pb2.BatchMapResponse]:
+        try:
+            async for msg in datum_iterator:
+                async for to_ret in self._process_one_flatmap(msg):
+                    yield to_ret
+        except Exception as err:
+            _LOGGER.critical("UDFError, re-raising the error", exc_info=True)
+            raise err
+
+    async def _process_one_flatmap(self, msg: Datum):
+        msg_id = msg.id
+
+        try:
+            # async for result in self._map_stream_handler.handler_stream(msg):
+            result = await self._handler(msg.keys, msg)
+            if isinstance(result, Message):
+                result = Messages(result)
+
+            yield batchmap_pb2.BatchMapResponse(
+                results=[
+                    batchmap_pb2.BatchMapResponse.Result(
+                        keys=msg.keys, value=msg.value, tags=msg.tags
+                    )
+                    for msg in result
+                ],
+                id=msg_id,
+            )
+
+        except Exception as err:
+            err_msg = "UDFError, re-raising the error: %r" % err
+            _LOGGER.critical(err_msg, exc_info=True)
+            raise err
+
+
+async def _make_async_iter(iterable):
+    for item in iterable:
+        yield item
+
+
+class BatchMapGroupingServicer(BatchMapServicer):
```

Comment:
Similar to the other comments, let's leave this functionality for a later change. Right now lets just support a single interface for BatchMap